packages/data/src/navigation/actions.js

```diff
@@ -89,6 +90,23 @@ export function removeFavoriteSuccess( favorite, error ) {
 	};
 }
 
+export function* onLoad() {
+	yield onHistoryChange();
+}
+
+export function* onHistoryChange() {
+	const persistedQuery = getPersistedQuery();
```

Comment:
Just making a note here (doesn't need to be fixed for this PR), in hindsight I guess the original `getPersistedQuery` should have been something more along the lines of `getPersistedQueryFromParams`.

This may be a tiny bit confusing going forward since we have `getPersistedQuery` in the data store as well as a util method to get from params.  But we can tackle this in a follow-up.